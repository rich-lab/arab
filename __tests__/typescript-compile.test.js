const { join, dirname } = require('path')
const typescriptCompile = require('../lib/typescript-compile')
const tsconfig = require('../lib/typescript/tsconfig')

describe('', () => {
  it('generate d.ts', () => {
    const filename = join(__dirname, 'fixtures/ts/export-interface.ts')

    const ret = typescriptCompile({
      fileNames: [filename],
      compilerOptions: tsconfig.compilerOptions
    })

    expect(ret).toMatchSnapshot()
  })

  it('generate d.ts for .tsx', () => {
    const filename = join(__dirname, 'fixtures/tsx/index.tsx')

    const ret = typescriptCompile({
      fileNames: [filename],
      compilerOptions: tsconfig.compilerOptions
    })

    expect(ret).toMatchSnapshot()
  })
})
