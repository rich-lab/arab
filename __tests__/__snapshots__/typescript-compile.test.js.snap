// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate d.ts 1`] = `
Object {
  "/Users/ulivz/Documents/ulivz/babe/__tests__/fixtures/ts/export-interface.ts": Object {
    "dts": "import { IFoo } from './interface';
export interface IFoo2 extends IFoo {
    title: string;
}
export declare function App(): void;
",
  },
}
`;

exports[`generate d.ts for .tsx 1`] = `
Object {
  "/Users/ulivz/Documents/ulivz/babe/__tests__/fixtures/tsx/index.tsx": undefined,
}
`;
